src/input/datetime.rs

```diff
@@ -220,10 +220,10 @@ impl<'a> EitherDateTime<'a> {
     pub fn try_into_py(self, py: Python<'a>) -> PyResult<PyObject> {
         let dt = match self {
             Self::Raw(datetime) => {
-                let tz: Option<PyObject> = match datetime.offset {
+                let tz: Option<PyTzInfo> = match datetime.offset {
```

Comment:
I think you want `Option<&PyTzInfo>` here (the PyO3 types are references, which is confusing and I hope to improve one day...)

Reply:
Ah yes probably thanks! But my main issue is to cast `TzInfo` as `PyTzInfo`
Since `TzInfo` extends `PyTzInfo` (```#[pyclass(module = "pydantic_core._pydantic_core", extends = PyTzInfo)]```)
I would have expected it to work